src/mbgl/map/annotation.cpp

```diff
@@ -234,8 +388,18 @@ std::vector<TileID> AnnotationManager::removeAnnotations(const AnnotationIDs& id
     return affectedTiles;
 }
 
-std::vector<uint32_t> AnnotationManager::getAnnotationsInBounds(const LatLngBounds& queryBounds,
-                                                                const MapData& data) const {
+const std::unique_ptr<Annotation>& AnnotationManager::getAnnotationWithID(uint32_t annotationID) const {
```

Comment:
Returning a reference is unlikely to be thread safe. What happens if the annotation is removed while another thread is using a reference to it?